keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1717,45 +1618,76 @@ type ResourceClaimTemplateSpec struct {
 }
 ```
 
-##### Object references
+##### ResourceQuota
+
+At the moment, this type is only used to control whether admin access is
+allowed in a namespace. Other aspects may get added later.
+
+Admin access can be checked in an admission plugin. Other limitations may have
+to be checked at allocation time, either because they depend on device
+attributes which cannot be assumed to be known at the time of admission of a
+claim or because the quota itself influences device selection. For example, a
+future "give me device A if available, otherwise device B" request might not be
+able to provide A because usage of A over quota while device B is still under
+quota.
 
 ```go
-// ResourceClassParametersReference contains enough information to let you
-// locate the parameters for a ResourceClass.
-type ResourceClassParametersReference struct {
-    // APIGroup is the group for the resource being referenced. It is
-    // empty for the core API. This matches the group in the APIVersion
-    // that is used when creating the resources.
-    // +optional
-    APIGroup string
-    // Kind is the type of resource being referenced. This is the same
-    // value as in the parameter object's metadata.
-    Kind string
-    // Name is the name of resource being referenced.
-    Name string
-    // Namespace that contains the referenced resource. Must be empty
-    // for cluster-scoped resources and non-empty for namespaced
-    // resources.
-    // +optional
-    Namespace string
+// Quota controls whether a ResourceClaim may get allocated.
+// Quota is namespaced and applies to claims within the same namespace.
+type Quota struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	metav1.ObjectMeta
+
+	Spec QuotaSpec
 }
 
-// ResourceClaimParametersReference contains enough information to let you
-// locate the parameters for a ResourceClaim. The object must be in the same
-// namespace as the ResourceClaim.
-type ResourceClaimParametersReference struct {
-    // APIGroup is the group for the resource being referenced. It is
-    // empty for the core API. This matches the group in the APIVersion
-    // that is used when creating the resources.
-    // +optional
-    APIGroup string
-    // Kind is the type of resource being referenced. This is the same
-    // value as in the parameter object's metadata, for example "ConfigMap".
-    Kind string
-    // Name is the name of resource being referenced.
-    Name string
+type QuotaSpec struct {
+	// AllowManagementAccess controls whether claims with ManagementAccess
+	// may be allocated. If multiple quota objects exist and at least one
+	// has a true value, access will be allowed. The default to deny such access.
+	//
+	// +optional
+	AllowManagementAccess bool `json:"allowManagementAccess,omitempty"`
```

Comment:
We called it "Admin" elsewhere. Suggest we pick one name...I think something more evocative of that actual behavior, as I mentioned in the previous comments.

Reply:
I fixed it to be consistent (i.e. "Admin access") and added some more explanations what that means.